src/utils/useAtomsSnapshot.ts

```diff
@@ -67,55 +67,57 @@ export function useAtomsSnapshot({
+          console.warn(
+            '[DEPRECATION-WARNING]: Your Jotai version is out-of-date and contains deprecated properties that will be removed soon. Please update to the latest version of Jotai.',
+          );
         }
 
-        const atomState = store.dev_get_atom_state?.(atom);
-        if (atomState) {
-          if ('v' in atomState) {
-            values.set(atom, atomState.v);
+        const values: AtomsValues = new Map();
+        const dependents: AtomsDependents = new Map();
+        for (const atom of store.dev_get_mounted_atoms?.() || []) {
+          if (!shouldShowPrivateAtoms && atom.debugPrivate) {
+            // Skip private atoms
+            continue;
           }
-        }
-        const mounted = store.dev_get_mounted?.(atom);
-        if (mounted) {
-          let atomDependents = mounted.t;
 
-          if (!shouldShowPrivateAtoms) {
-            // Filter private dependent atoms
-            atomDependents = new Set(
-              Array.from(atomDependents.values()).filter(
-                /* NOTE: This just removes private atoms from the dependents list,
-                 instead of hiding them from the dependency chain and showing
-                 the nested dependents of the private atoms. */
-                (dependent) => !dependent.debugPrivate,
-              ),
-            );
+          const atomState = store.dev_get_atom_state?.(atom);
+          if (atomState) {
+            if ('v' in atomState) {
+              values.set(atom, atomState.v);
+            }
           }
+          const mounted = store.dev_get_mounted?.(atom);
+          if (mounted) {
+            let atomDependents = mounted.t;
+
+            if (!shouldShowPrivateAtoms) {
+              // Filter private dependent atoms
+              atomDependents = new Set(
+                Array.from(atomDependents.values()).filter(
+                  /* NOTE: This just removes private atoms from the dependents list,
+                  instead of hiding them from the dependency chain and showing
+                  the nested dependents of the private atoms. */
+                  (dependent) => !dependent.debugPrivate,
+                ),
+              );
+            }
 
-          dependents.set(atom, atomDependents);
+            dependents.set(atom, atomDependents);
+          }
+        }
+        if (
+          isEqualAtomsValues(prevValues, values) &&
+          isEqualAtomsDependents(prevDependents, dependents)
+        ) {
+          // not changed
+          return;
         }
-      }
-      if (
-        isEqualAtomsValues(prevValues, values) &&
-        isEqualAtomsDependents(prevDependents, dependents)
-      ) {
-        // not changed
-        return;
-      }
-      prevValues = values;
-      prevDependents = dependents;
-      setAtomsSnapshot({ values, dependents });
+        prevValues = values;
+        prevDependents = dependents;
+        setAtomsSnapshot({ values, dependents });
+      });
```

Comment:
I think we should only wrap the setter:
```js
Promise.resolve().then(() => {
  setAtomsSnapshot({ values, dependents });
});
```

Though, on second thought, it might be better to detect if it's sync, and only delay if necessary.